File: youzi_v2/internal_tracking.py
"""内部路由轨迹占位文案（视为无有效轨迹）。"""

from __future__ import annotations

INTERNAL_WAREHOUSE_PLACEHOLDER = "Your goods are in the warehouse"


def is_internal_no_tracking_desc(desc: str | None) -> bool:
    return (desc or "").strip() == INTERNAL_WAREHOUSE_PLACEHOLDER


def mask_internal_summary(time: str | None, desc: str | None) -> tuple[str, str]:
    if is_internal_no_tracking_desc(desc):
        return "", ""
    return (time or "").strip(), (desc or "").strip()

File: youzi_v2/frontend/src/utils/internalTracking.ts
/** 与后端 internal_tracking.INTERNAL_WAREHOUSE_PLACEHOLDER 一致 */
export const INTERNAL_WAREHOUSE_PLACEHOLDER = 'Your goods are in the warehouse'

export function isInternalNoTrackingDesc(desc?: string | null): boolean {
  return (desc ?? '').trim() === INTERNAL_WAREHOUSE_PLACEHOLDER
}

export function hasEffectiveInternalTracking(row: {
  latestTrackingTime?: string | null
  latestTrackingDesc?: string | null
}): boolean {
  if (isInternalNoTrackingDesc(row.latestTrackingDesc)) return false
  return Boolean(row.latestTrackingTime?.trim() || row.latestTrackingDesc?.trim())
}
